src/server/lib/middleware/requestState.ts

```diff
@@ -0,0 +1,28 @@
+// Middleware for assigning important parts of the request state to res.locals.
+// This should be the only place res.locals is mutated.
+// Requires: csurf middlware
+import { getGeolocationRegion } from '@/server/lib/getGeolocationRegion';
+import { getDefaultRequestState } from '@/server/models/Express';
+import { parseExpressQueryParams } from '@/server/lib/queryParams';
+import { NextFunction, Request, Response } from 'express';
+import { getConfiguration } from '../getConfiguration';
+import { tests } from '@/shared/model/experiments/abTests';
+import { getABTesting } from '../getABTesting';
+
+export const requestStateMiddleware = (
+  req: Request,
+  res: Response,
+  next: NextFunction,
+) => {
+  const state = getDefaultRequestState();
```

Comment:
Can we avoid mutating an object returned by `getDefaultRequestState`?
Instead we could have a function `createDefaultRequestState` that takes the middleware state as parameters, and sets them on object creation.

Reply:
Yes, that's the plan for when I implement the immutability linting (I'll have to do this for other bits of code too)

Reply:
👍  I pointed it how here because this is new code.